File: node/wags-media-repository-web/src/models/VideoGame.ts
export type VideoGameSystem = {
    videoGameSystemId: number;
    name: string;
    colorCode: string;
    videoGameCount?: number;
}

export type VideoGameGenre = {
    videoGameGenreId: number;
    name: string;
    colorCode: string;
    videoGameCount?: number;
}

export type VideoGameStatus = {
    videoGameStatusId: number;
    name: string;
    colorCode: string;
}

export type VideoGameCompletion = {
    videoGameCompletionId: number;
    name: string;
    colorCode: string;
}

export type VideoGame = {
    videoGameId: number;
    videoGameStatusId: number;
    videoGameCompletionId: number;
    title: string;
    link: string;
    dateStarted?: Date;
    dateCompleted?: Date;
    rating: number;
    thoughts: string;
    coverImageUrl: string;
    sortOrder?: number;
    status?: VideoGameStatus;
    completion?: VideoGameCompletion;
    genres: VideoGameGenre[];
    systems: VideoGameSystem[];
}

File: tools/WagsMediaGenerator/Models/VideoGame.cs
namespace WagsMediaGenerator.Models;

public class VideoGame
{
    public int VideoGameId { get; set; }
    
    public string Title { get; set; } = string.Empty;
    
    public string Link { get; set; } = string.Empty;
    
    public DateTime? DateStarted { get; set; }
    
    public DateTime? DateCompleted { get; set; }
    
    public int Rating { get; set; }

    public string Thoughts { get; set; } = string.Empty;
    
    public string CoverImageUrl { get; set; } = string.Empty;
    
    public int? SortOrder { get; set; }

    public Tag Status { get; set; } = new();

    public Tag Completion = new();

    public List<Tag> Genres { get; set; } = [];
    
    public List<Tag> Systems { get; set; } = [];
}